### DISCUSSION TITLE
[V5] Adonis contracts

### DISCUSSION THREAD

**Discussion by tridungle** [adonisjs/core#2686] [Help] [answered]
Can I define my own interface in the contracts folder? If I can, how to do it

**Comment by arthurcvm** [comment]
Please be more specific.

**Reply by tridungle** [reply]
@arthurcvm Thank you for your reply.
For example:
in the contracts folder, I create  a file common.ts and declare my interface here:
`declare module '@ioc:MyModule/Contracts/Common' {
    interface Something {}
}`
I assume that I can call this module anywhere in my app, but I got an error:
`E_IOC_LOOKUP_FAILED: Cannot resolve "MyModule/Contracts/Common" namespace from the IoC Container`

**Reply by scal-lider** [reply]
How do you using it throw your code?

**Reply by tridungle** [reply]
@scal-lider 
`import { Something } from '@ioc:MyModule/Contracts/Common'`

**Reply by arthurcvm** [reply]
In my case I created a "protocols" folder in the app and put the interfaces inside, but I don't know if is the best solution.

**Comment by d4n1b** [comment, marked as answer]
Hi @tridungle ,
Here's an [example](https://docs.adonisjs.com/guides/security/hashing#updating-the-contract) of how to update an existing contract. Do notice that it's just for typing purposes.

The error you're seeing is an issue with the path you're using to import your new module. If required, you can add custom resolvers in `compilerOptions.paths` in `tsconfig.json`

```
{
  "extends": "./node_modules/adonis-preset-ts/tsconfig",
  "include": [
    "**/*"
  ],
  "exclude": [
    "node_modules",
    "build"
  ],
  "compilerOptions": {
    // ...
    "paths": {
      "App/*": [
        "./app/*"
      ],
      "Config/*": [
        "./config/*"
      ],
      "Contracts/*": [
        "./contracts/*"
      ],
      "Database/*": [
        "./database/*"
      ]
    },
```

I hope it helps.

**Reply by d4n1b** [reply]
@tridungle,

Contracts are meant to be for type purposes, you might just need to change `'@ioc:MyModule/Contracts/Common'` with `'MyModule/Contracts/Common'`, as AFAIK, `@ioc:` folder is meant to be for internal AdonisJS packages.

**Reply by d4n1b** [reply]
Hey @tridungle,
If any of the answers provide a solution, do you mind flagging the thread as "answered"?

Thanks 😊
